src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -262,26 +274,47 @@ public ResponseEntity<byte[]> getCourseIcon(@PathVariable Long courseId, @PathVa
     }
 
     /**
-     * GET /files/attachments/access-token/{filename:.+} : Generates an access token that is valid for 30 seconds and given filename
+     * GET files/file-upload-exercises/:exerciseId/submissions/:submissionId/:filename/access-token : Generates an access token that is valid for 30 seconds and given exerciseId and submissionId
      *
-     * @param filename name of the file, the access token is for
+     * @param exerciseId ID of the exercise the submission belongs to
+     * @param submissionId ID of the submission the access token is for
+     * @param filename The filename of the file
      * @return The generated access token
      */
-    @GetMapping("files/attachments/access-token/{filename:.+}")
+    @GetMapping("files/file-upload-exercises/{exerciseId}/submissions/{submissionId}/{filename:.+}/access-token")
     @PreAuthorize("hasRole('USER')")
-    public ResponseEntity<String> getTemporaryFileAccessToken(@PathVariable String filename) {
-        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
-        // TODO: we should check that the filename actually exists in the context in Artemis, ideally we also verify that the user has access to this file already here
-        if (filename == null) {
+    public ResponseEntity<String> getTemporaryFileUploadExerciseSubmissionAccessToken(@PathVariable Long exerciseId, @PathVariable Long submissionId,
+            @PathVariable String filename) {
+        log.debug("REST request to get an access_token for a file upload exercise submission with exerciseId: {} and submissionId {}", exerciseId, submissionId);
+
+        Optional<FileUploadSubmission> optionalSubmission = fileUploadSubmissionRepository.findById(submissionId);
```

Comment:
Threre should be `findByIdOrElseThrow` (or somewhat similar) methods availibe here. That's the case for basicly all calls to a repository in these changes.

Reply:
I looked into your suggestion, but using `findByIdOrElseThrow` here and in the other calls would then require me to catch the thrown exceptions, otherwise the endpoint would leak internals.
Or am I missing something?

Reply:
I don't think that this is a problem, we usually just throw the exception, see for example the changes in #4651